backend/api/v1/endpoints/Auth/auth.py

```
@@ -0,0 +1,101 @@
+        
+        if not username:
+            raise HTTPException(status_code=401, detail="Invalid token")
+        
+        user = db.query(models.User).filter(models.User.username == username).first()
+        if not user:
+            raise HTTPException(status_code=404, detail= user_not_found)
+        
+        return user
+    
+    except jwt.ExpiredSignatureError:
+        raise HTTPException(status_code=401, detail="Token expired")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)